app/task_routes.py

```diff
@@ -0,0 +1,163 @@
+from app import db
+from app.models.task import Task
+from app.models.goal import Goal
+from flask import Blueprint, jsonify, abort, make_response, request
+import requests
+from datetime import * 
+import os
+
+
+# register task bp
+task_bp = Blueprint("task_bp", __name__, url_prefix="/tasks")
+SLACK_BOT_TOKEN = os.environ.get('SLACK_BOT_TOKEN')
+
+# validate task by id!
+def validate_model(cls, model_id):
+    try:
+        model_id = int(model_id)
+    except:
+        abort(make_response({"message":f"{cls.__name__} {model_id} invalid"}, 400))
+
+    model = cls.query.get(model_id)
+
+    if not model:
+        abort(make_response({"message":f"{cls.__name__} {model_id} not found"}, 404))
+
+    return model
+
+# create task as a post request 
+@task_bp.route("",methods=["POST"])
+def create_task():
+    request_body = request.get_json()
+
+    #try except
+    try:
+        task = Task.from_dict(request_body)
+    except:
+        return {"details": "Invalid data"}, 400
+
+    db.session.add(task)
+    db.session.commit()
+
+    return {"task":{
+            "id": task.task_id,
+            "description": task.description,
+            "is_complete": (task.completed_at != None),
+            "title": task.title}}, 201
+
+
+# read all tasks 
+@task_bp.route("",methods=["GET"])
+def read_all_tasks():
+    tasks = Task.query.all()
+
+    # ascending sort
+    sorting_query = request.args.get("sort")
+    if sorting_query == "asc":
+        tasks = Task.query.order_by(Task.title.asc()).all()
+    elif sorting_query == "desc":
+        tasks = Task.query.order_by(Task.title.desc()).all()
+    else:
+        tasks = Task.query.all()
+
+    tasks_response = []
+    for task in tasks:
+        tasks_response.append(task.to_dict())
+    
+    return jsonify(tasks_response), 200
+
+
+# read one task
+@task_bp.route("/<task_id>", methods=["GET"])
+def read_one_task(task_id):
+    task = validate_model(Task, task_id)
+    if task.goal_id:
+        return {
+            "task":{
+                "id": task.task_id,
+                "title": task.title,
+                "description": task.description,
+                "goal_id": task.goal_id, 
+                "is_complete": (task.completed_at != None)
+            }
+        }
+    else:
```

Comment:
Another way you could implement this by changing your `to_dict` method for Task to include a `goal_id` if one is present.